### PR TITLE
web: migrate Bootstrap button classes to the Wildcard component

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

Migration of the outstanding Bootstrap button classes to the Wildcard `Button` component.

## Notes

- Most of the diff comes from updated snapshots extracted into a separate commit. 🙂
- The migration of the shared components required disabling branded styles in the `Button` component, which required adding the `branded` context to unit tests to avoid changing snapshots. It's done in one standalone commit: [`web: update snapshots`](https://github.com/sourcegraph/sourcegraph/pull/30621/commits/73b87db28b662c2de30b0bb9004cd3d11f1324e0).

Follow-up for https://github.com/sourcegraph/sourcegraph/pull/30096.
Closes https://github.com/sourcegraph/sourcegraph/issues/29318.

### CODE REVIEW

**Comment by valerybugakov** [comment]
@cla-bot check

File: client/vscode/src/webview/search-sidebar/index.tsx [hunk 1]
```diff
@@ -63,7 +69,11 @@ const Main: React.FC = () => {
     // TODO: should we hide the access token form permanently if an unauthenticated user
     // has performed a search before? Or just for this session?
     if (state.status === 'search-home' && !authenticatedUser) {
-        return <AuthSidebarView {...webviewPageProps} />
+        return (
+            <WildcardThemeContext.Provider value={{ isBranded: true }}>
```
Comment (valerybugakov) [inline client/vscode/src/webview/search-sidebar/index.tsx:73]: @tjkandala, added `WildcardThemeContext` here required for `AuthSidebarView` buttons.

File: client/web/src/search/render-branded-to-string.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Helper used to render components to string with branded styles.
+ * Required to render Wildcard components as `text` field with `shepherd.js` library.
+ * See related issue here: https://github.com/shipshapecode/react-shepherd/issues/612.
+ */
+export function renderBrandedToString(children: React.ReactNode): string {
+    return ReactDOMServer.renderToString(
+        <WildcardThemeContext.Provider value={wildcardTheme}>{children}</WildcardThemeContext.Provider>
+    )
+}
```
Comment (valerybugakov) [inline client/web/src/search/render-branded-to-string.tsx:19]: This helper is required to use React components in the search tour.

**Review by umpox — APPROVED** [review]
👍

**Pull request merged**